### ISSUE TITLE
BUG: breaking change in df.replace() from 1.0.5 to 1.1.0

### ISSUE THREAD

**Issue by agnesbao** [pandas-dev/pandas#35931]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
# Your code here
df = pd.DataFrame({"age": range(1,100)})
df["age"] = pd.cut(df["age"], [0, 50, 65, 99], right=False)
df.replace(to_replace={
    "age":{
        pd.Interval(0, 50, closed='left'): "Ages < 50",
        pd.Interval(50, 65, closed='left'): "Ages 50-64",
        pd.Interval(65, 99, closed='left'): "Ages 65+"
    }
})
```

#### Problem description

The above sample code works in 1.0.5 but error out in 1.1.0 and above with error msg:
```
TypeError: Cannot compare types 'ndarray(dtype=object)' and 'Interval'
```
It only happens with pd.Interval dtype, nested dict of string to string mapping works fine. 

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : e3bcf8d1949833793ff50cca393240dab0f0d461
python           : 3.7.6.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.6.0
Version          : Darwin Kernel Version 19.6.0: Thu Jun 18 20:49:00 PDT 2020; root:xnu-6153.141.1~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.2.0.dev0+165.ge3bcf8d19
numpy            : 1.19.1
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.2
setuptools       : 49.6.0.post20200814
Cython           : 0.29.21
pytest           : 6.0.1
hypothesis       : 5.29.0
sphinx           : 2.4.0
blosc            : None
feather          : None
xlsxwriter       : 1.3.3
lxml.etree       : 4.5.2
html5lib         : 1.1
pymysql          : None
psycopg2         : 2.8.4 (dt dec pq3 ext lo64)
jinja2           : 2.11.2
IPython          : 7.17.0
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fsspec           : 0.8.0
fastparquet      : 0.4.1
gcsfs            : None
matplotlib       : 3.3.1
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.5
pandas_gbq       : None
pyarrow          : 0.15.1
pytables         : None
pyxlsb           : None
s3fs             : 0.2.0
scipy            : 1.5.2
sqlalchemy       : 1.3.19
tables           : 3.6.1
tabulate         : 0.8.7
xarray           : 0.16.0
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : 0.50.1

</details>

**Comment by asishm** [comment]
#32107


> c6e3e158a0e361a918e47bb0db12e34c46bac998 is the first bad commit
> commit c6e3e158a0e361a918e47bb0db12e34c46bac998
> Author: Daniel Saxton <2658661+dsaxton@users.noreply.github.com>
> Date:   Sat Apr 25 18:58:51 2020 -0500
> 
>     BUG: Allow addition of Timedelta to Timestamp interval (#32107)

**Comment by dsaxton** [comment]
Hmm, this is apparently due to https://github.com/pandas-dev/pandas/blob/a1f605697e44e7668b64841baf4ab2988470d08c/pandas/_libs/interval.pyx#L302

which was added to allow np.timedelta64 + pd.Interval of timestamps in the linked PR: https://github.com/pandas-dev/pandas/pull/32107#discussion_r407097250. Not sure exactly what's happening to be honest but will take a look.

cc @jbrockmendel if any thoughts

**Comment by dsaxton** [comment]
So evidently `__array_priority__ = 1000` has the not so nice side effect of no longer broadcasting things like equality comparisons (which is causing the bug in the OP):
```python
In [1]: import numpy as np
   ...: import pandas as pd
   ...:
   ...: arr = np.array([pd.Interval(0, 1), pd.Interval(1, 2)])
   ...: arr == pd.Interval(0, 1)
   ...:
Out[1]: False
```
Probably this behavior is worse than not being able to add np.timedelta64 + pd.Interval(pd.Timestamp, pd.Timestamp).

**Comment by simonjayhawkins** [comment]
fixed in #35938
